README.md

```diff
@@ -1 +1,76 @@
-# PSP22
\ No newline at end of file
+# PSP22 Fungible Token
+
+PSP22 is a fungible token standard for WebAssembly smart contracts running on blockchains based on the [Substrate][substrate] framework. It is an equivalent of Ethereum's [ERC-20][erc20]. The definition of the PSP22 standard can be found [here][psp22].
+
+This repository contains a simple, minimal implementation of the PSP22 token in [ink!][ink] programming language.
+
+## How to use this repository
+
+To use this crate please add the following line in your `Cargo.toml`:
+```
+psp22 = { git = "https://github.com/Cardinal-Cryptography/PSP22.git", default-features = false }
+```
+
+The contents of this repository can be used in following ways:
+
+### 1. Cross contract calling with traits
+
+The `PSP22` trait contains all the methods defined in the PSP22 standard. The trait can be used together with ink!'s [`contract_ref`][contract_ref] macro to allow for convenient cross-contract calling.
+
+In your contract, if you would like to make a call to some other contract implementing the PSP22 standard, all you need to do is:
+```
+use ink::contract_ref;
+use psp22::PSP22;
+
+let mut token: contract_ref!(PSP22) = other_address.into();
+
+// Now `token` has all the PSP22 methods
+let balance = token.balance_of(some_account);
+token.transfer(recipient, value); // returns Result<(), PSP22Error>
+```
+
+The same method can be used with other traits (`PSP22Metadata`, `PSP22Burnable`, `PSP22Mintable`) defined in this crate. See the contents of [`traits.rs`][traits].
+
+### 2. Ready to use contract
```

Comment:
Nit: I'd kind of consider putting this as the first use case, as I believe it will be the most popular one.

Reply:
Good idea, done